_overviews/scala3-book/domain-modeling-tools.md

```diff
@@ -0,0 +1,575 @@
+---
+title: Tools
+type: section
+description: This chapter provides an introduction to the available domain modeling tools in Scala 3, including classes, traits, enums, and more.
+num: 20
+previous-page: domain-modeling-intro
+next-page: domain-modeling-oop
+---
+
+
+
+Scala 3 provides the following tools to help us model the world around us:
+
+- Class
+- Companion object
+- Trait
+- Abstract class
+- Enum
+- Case class
+- Case object
+
+The following sections introduce each of these modeling tools.
+
+
+
+## Classes
+
+As with other languages, a Scala *class* is a template for the creation of object instances. Here are some examples of classes that have constructor parameters, but no fields or methods:
```

Comment:
IIRC specifying `var` means that you are creating fields in the class. To only have constructor parameters, but no fields or methods, you should drop the `var`s

Reply:
@alvinj if you change the code to address @vincenzobaz comment, then also the text following needs adjustment since it talks about var fields. In fact,
```scala
class A(var x: T)
```
creates a field `x` and a constructor expecting a `T` and assigning to `x`.

Reply:
This was re-worded, so I’m resolving this conversation.